src/btree/bt_split.c

```diff
@@ -2360,6 +2360,15 @@ __wt_split_rewrite(WT_SESSION_IMPL *session, WT_REF *ref, WT_MULTI *multi)
 
     __wt_ref_out(session, ref);
 
+    /*
+     * It is possible for disaggregated storage to do a split write for the internal page. Update
+     * the parent ref on the page as we used a dummy ref to instantiate the page.
+     */
+    if (F_ISSET(ref, WT_REF_FLAG_INTERNAL)) {
+        WT_ASSERT(session, F_ISSET(S2BT(session), WT_BTREE_DISAGGREGATED));
```

Comment:
I think we may also hit this in develop. It is just very unlikely as we should have turned off scrub eviction at the time when we start to evict internal pages.